asciidoc/modules/con_mg_changes-in-key-management.adoc

```diff
@@ -0,0 +1,121 @@
+[id="changes-in-key-management_{context}"]
+= Changes in key management
+
+In {VertX} {4}, there are major updates in handling keys. The most important change is that when a key loads, there is no distinction between public buffer and private buffer.
+
+The following classes have been updated:
+
+* `io.vertx.ext.auth.KeyStoreOptions` used to work with `jce` keystores
+
+* `io.vertx.ext.auth.SecretOptions` used to handle symmetric secrets
+
+* `io.vertx.ext.auth.PubSecKeyOptions` used to handle public secret keys
+
+The following section describes the changes in key management.
+
+== Secret options class is no longer available
+
+The `SecretOptions` class is no longer available. Use the new `PubSecKeyOptions` class instead to work with a cryptographic key.
+
+The following example shows how methods of `SecretOptions` class were used in {VertX} {3x} releases.
+
+----
+new SecretOptions()
+    .setType("HS256")
+    .setSecret("password")
+----
+
+The following example shows how methods of `PubSecKeyOptions` class should be used in {VertX} {4}.
+
+----
+new PubSecKeyOptions()
+    .setAlgorithm("HS256")
+    .setSecretKey("password")
+----
+
+== Updates in public secret keys management
+
+In {VertX} {3x}, the configuration object in public secret key management assumed that:
+
+* Keys are configured as key-pairs.
+* Key data is a PKCS8 encoded string without standard delimiters.
+
+The following example shows how to configure key pair in {VertX} {3x}.
+
+[source,java,options="nowrap",subs="attributes+"]
+----
+new PubSecKeyOptions()
+  .setPublicKey(
+    // remove the boundaries
+    // (don't do this as it doesn't handle end of lines
+    // it's for illustration purposes)
+    pubPemString
+      .replaceAll("-----BEGIN PUBLIC KEY----")
+      .replaceAll("-----END PUBLIC KEY----"))
+  .setSecretKey(
+    // remove the boundaries
+    // (don't do this as it doesn't handle end of lines
+    // it's for illustration purposes)
+    secPemString
+      .replaceAll("-----BEGIN PUBLIC KEY----")
+      .replaceAll("-----END PUBLIC KEY----"));
+----
+
+In {VertX} {4}, the configuration uses a single key, so a key-pair should be 2 key (the public
+the secret)XXX.
```

Comment:
@vietj Can you please help me understand this sentence?

Reply:
@pmlopes ping

Reply:
@sangeetaraghu In vert.x 4 the configuration has been simplified to handle just `Key`s. There are 2 kinds of keys: `public` and `private`. Before a custom extra kind was also available `key-pair`. Since a key pair is just as the name implies a `public` plus a `private` key, they now need to be specified independently.

Reply:
@pmlopes I have rephrased the sentence. Will this work?

**In {VertX} {4}, you must specify both the public and private key.**

Removed this sentence:
In {VertX} {4}, the configuration uses a single key, so a key-pair should be 2 key (the public
the secret)XXX.